### ISSUE TITLE
Renovate PRs failing to update artifact with error about `--ignore-platform`

### ISSUE THREAD

**Issue by justinhaaheim** [renovatebot/renovate#16535]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

Just installed Renovate. Renovate PRs are failing on my repo with the following error:

```
⚠ Artifact update problem
Renovate failed to update an artifact related to this branch. You probably do not want to merge this PR as-is.

♻ Renovate will retry this branch, including artifacts, only when one of the following happens:

any of the package files in this branch needs updating, or
the branch becomes conflicted, or
you click the rebase/retry checkbox if found above, or
you rename this PR's title to start with "rebase!" to trigger it manually
The artifact failure details are included below:

File name: yarn.lock
installing v2 tool yarn-slim v1.22.19
npm WARN config global `--global`, `--local` are deprecated. Use `--location=global` instead.

added 1 package in 1s
linking tool yarn-slim v1.22.19
3.2.1
Unknown Syntax Error: Unsupported option name ("--ignore-platform").

$ yarn install [--json] [--immutable] [--immutable-cache] [--check-cache] [--inline-builds] [--mode #0]
```

### Relevant debug logs

<details><summary>Logs</summary>

```

DEBUG: Executing command(branch="renovate/socket.io-packages")
{
  "command": "docker run --rm --name=renovate_node --label=renovate_child -v \"/mnt/renovate/gh/justinhaaheim/phrase-game\":\"/mnt/renovate/gh/justinhaaheim/phrase-game\" -v \"/tmp/renovate-cache\":\"/tmp/renovate-cache\" -e NPM_CONFIG_CACHE -e npm_config_store -e CI -e YARN_CACHE_FOLDER -w \"/mnt/renovate/gh/justinhaaheim/phrase-game\" docker.io/renovate/node:16.16.0 bash -l -c \"install-tool yarn-slim 1.22.19 && yarn install --ignore-engines --ignore-platform --network-timeout 100000 --ignore-scripts\""
}
DEBUG: rawExec err(branch="renovate/socket.io-packages")
{
  "err": {
    "killed": false,
    "code": 1,
    "signal": null,
    "cmd": "docker run --rm --name=renovate_node --label=renovate_child -v \"/mnt/renovate/gh/justinhaaheim/phrase-game\":\"/mnt/renovate/gh/justinhaaheim/phrase-game\" -v \"/tmp/renovate-cache\":\"/tmp/renovate-cache\" -e NPM_CONFIG_CACHE -e npm_config_store -e CI -e YARN_CACHE_FOLDER -w \"/mnt/renovate/gh/justinhaaheim/phrase-game\" docker.io/renovate/node:16.16.0 bash -l -c \"install-tool yarn-slim 1.22.19 && yarn install --ignore-engines --ignore-platform --network-timeout 100000 --ignore-scripts\"",
    "stdout": "installing v2 tool yarn-slim v1.22.19\nnpm WARN config global `--global`, `--local` are deprecated. Use `--location=global` instead.\n\nadded 1 package in 1s\nlinking tool yarn-slim v1.22.19\n3.2.1\nUnknown Syntax Error: Unsupported option name (\"--ignore-platform\").\n\n$ yarn install [--json] [--immutable] [--immutable-cache] [--check-cache] [--inline-builds] [--mode #0]\n",
    "stderr": "",
    "message": "Command failed: docker run --rm --name=renovate_node --label=renovate_child -v \"/mnt/renovate/gh/justinhaaheim/phrase-game\":\"/mnt/renovate/gh/justinhaaheim/phrase-game\" -v \"/tmp/renovate-cache\":\"/tmp/renovate-cache\" -e NPM_CONFIG_CACHE -e npm_config_store -e CI -e YARN_CACHE_FOLDER -w \"/mnt/renovate/gh/justinhaaheim/phrase-game\" docker.io/renovate/node:16.16.0 bash -l -c \"install-tool yarn-slim 1.22.19 && yarn install --ignore-engines --ignore-platform --network-timeout 100000 --ignore-scripts\"\n",
    "stack": "Error: Command failed: docker run --rm --name=renovate_node --label=renovate_child -v \"/mnt/renovate/gh/justinhaaheim/phrase-game\":\"/mnt/renovate/gh/justinhaaheim/phrase-game\" -v \"/tmp/renovate-cache\":\"/tmp/renovate-cache\" -e NPM_CONFIG_CACHE -e npm_config_store -e CI -e YARN_CACHE_FOLDER -w \"/mnt/renovate/gh/justinhaaheim/phrase-game\" docker.io/renovate/node:16.16.0 bash -l -c \"install-tool yarn-slim 1.22.19 && yarn install --ignore-engines --ignore-platform --network-timeout 100000 --ignore-scripts\"\n\n    at ChildProcess.exithandler (child_process.js:383:12)\n    at ChildProcess.emit (events.js:400:28)\n    at ChildProcess.emit (domain.js:475:12)\n    at maybeClose (internal/child_process.js:1058:16)\n    at Process.ChildProcess._handle.onexit (internal/child_process.js:293:5)"
  }
}
DEBUG: lock file error(branch="renovate/socket.io-packages")
{
  "err": {
    "killed": false,
    "code": 1,
    "signal": null,
    "cmd": "docker run --rm --name=renovate_node --label=renovate_child -v \"/mnt/renovate/gh/justinhaaheim/phrase-game\":\"/mnt/renovate/gh/justinhaaheim/phrase-game\" -v \"/tmp/renovate-cache\":\"/tmp/renovate-cache\" -e NPM_CONFIG_CACHE -e npm_config_store -e CI -e YARN_CACHE_FOLDER -w \"/mnt/renovate/gh/justinhaaheim/phrase-game\" docker.io/renovate/node:16.16.0 bash -l -c \"install-tool yarn-slim 1.22.19 && yarn install --ignore-engines --ignore-platform --network-timeout 100000 --ignore-scripts\"",
    "stdout": "installing v2 tool yarn-slim v1.22.19\nnpm WARN config global `--global`, `--local` are deprecated. Use `--location=global` instead.\n\nadded 1 package in 1s\nlinking tool yarn-slim v1.22.19\n3.2.1\nUnknown Syntax Error: Unsupported option name (\"--ignore-platform\").\n\n$ yarn install [--json] [--immutable] [--immutable-cache] [--check-cache] [--inline-builds] [--mode #0]\n",
    "stderr": "",
    "message": "Command failed: docker run --rm --name=renovate_node --label=renovate_child -v \"/mnt/renovate/gh/justinhaaheim/phrase-game\":\"/mnt/renovate/gh/justinhaaheim/phrase-game\" -v \"/tmp/renovate-cache\":\"/tmp/renovate-cache\" -e NPM_CONFIG_CACHE -e npm_config_store -e CI -e YARN_CACHE_FOLDER -w \"/mnt/renovate/gh/justinhaaheim/phrase-game\" docker.io/renovate/node:16.16.0 bash -l -c \"install-tool yarn-slim 1.22.19 && yarn install --ignore-engines --ignore-platform --network-timeout 100000 --ignore-scripts\"\n",
    "stack": "Error: Command failed: docker run --rm --name=renovate_node --label=renovate_child -v \"/mnt/renovate/gh/justinhaaheim/phrase-game\":\"/mnt/renovate/gh/justinhaaheim/phrase-game\" -v \"/tmp/renovate-cache\":\"/tmp/renovate-cache\" -e NPM_CONFIG_CACHE -e npm_config_store -e CI -e YARN_CACHE_FOLDER -w \"/mnt/renovate/gh/justinhaaheim/phrase-game\" docker.io/renovate/node:16.16.0 bash -l -c \"install-tool yarn-slim 1.22.19 && yarn install --ignore-engines --ignore-platform --network-timeout 100000 --ignore-scripts\"\n\n    at ChildProcess.exithandler (child_process.js:383:12)\n    at ChildProcess.emit (events.js:400:28)\n    at ChildProcess.emit (domain.js:475:12)\n    at maybeClose (internal/child_process.js:1058:16)\n    at Process.ChildProcess._handle.onexit (internal/child_process.js:293:5)"
  },
  "type": "yarn"
}
DEBUG: No updated lock files in branch(branch="renovate/socket.io-packages")
DEBUG: PR has no releaseTimestamp(branch="renovate/socket.io-packages")
DEBUG: 1 file(s) to commit(branch="renovate/socket.io-packages")
```

</details>

### Have you created a minimal reproduction repository?

No reproduction repository

**Comment by rarkins** [comment]
Which version of Yarn are you using in your repo and how is it set up?

**Comment by adam-h** [comment]
We're seeing this same issue, yarn 3.1. Looks like this was an undocumented flag in yarn 1.x, but removed in yarn > 2.

**Comment by MartijnLammaing** [comment]
Hi there, 

Hopefully this will shed some light into the issue.

https://github.com/MartijnLammaing/RenovateYarnIssueReproduction

Hopefully the minimal reproduction is done properly enough.

**Comment by viceice** [comment]
ok, you're mixing yarn V1 and V2+. please add the `packageManager` field to your package.json to support corepack and your happy to get it working

**Comment by MartijnLammaing** [comment]
Hey @viceice, thanks for your suggestion! We had a reason why we kept the package manager to 1.19 though, which now kinda eludes me. I'll try it, thanks!

**Comment by justinhaaheim** [comment]
> ok, you're mixing yarn V1 and V2+. please add the `packageManager` field to your package.json to support corepack and your happy to get it working

This fixed my issue. packageManager was specified in the root of the monorepo, but not in the package.json files for each of the workspaces.

Would it make sense to warn here when different versions of a package manager are being used by different parts of a monorepo (unintentionally) to try to make this issue more obvious?

**Comment by rarkins** [comment]
I have forked the reproduction by @MartijnLammaing to https://github.com/renovate-reproductions/16535

However when I look at the PR in the original reproduction repo, I see no artifacts error: https://github.com/MartijnLammaing/RenovateYarnIssueReproduction/pull/5

It also doesn't sound exactly the same as @justinhaaheim found, because there's no monorepo in the reproduction.

@justinhaaheim are you able to reproduce your monorepo problem separately to test with?

**Comment by haakonflatval-cognite** [comment]
To anyone else experiencing this, I was able to work around this using
```
{
  ...
  "force": {
    "constraints": {
      "yarn": ">= 3.0.0"
    }
  }
  ```
at the end of the `renovate` JSON file. It seems to make `renovate` use `yarn` arguments compatible with newer Yarn versions.

**Comment by viceice** [comment]
@haakonflatval-cognite you should use `packageManager` in `package.json` field instead

**Comment by Lonli-Lokli** [comment]
I have such field https://gitlab.com/lonli.lokli/cv-app/-/blob/develop/frontend/cv-ui/package.json#L125
But still have same error https://gitlab.com/lonli.lokli/cv-app/-/merge_requests/494
```
Unknown Syntax Error: Unsupported option name ("--ignore-platform").

$ yarn install [--json] [--immutable] [--immutable-cache] [--check-cache] [--inline-builds] [--mode #0]

```

**Comment by rarkins** [comment]
Duplicate of #20509

Should be fixed in 34.146.1
